cloudbridge/cloud/providers/gce/services.py

```diff
@@ -872,3 +1024,76 @@ def create(self, name, volume, description=None):
             return snapshots[0]
         else:
             return None
+
+
+class GCSObjectStoreService(BaseObjectStoreService):
+
+    def __init__(self, provider):
+        super(GCSObjectStoreService, self).__init__(provider)
+
+    def get(self, bucket_id):
+        """
+        Returns a bucket given its ID. Returns ``None`` if the bucket
+        does not exist or if the user does not have permission to access the
+        bucket.
+        """
+        try:
+            response = (self.provider
+                            .gcp_storage
+                            .buckets()
+                            .get(bucket=bucket_id)
+                            .execute())
+            if 'error' in response:
```

Comment:
Should this check for 403 and 404 and otherwise raise an exception? log an error?